frontend/src/metabase/data-grid/components/DataGrid/DataGrid.tsx

```diff
@@ -195,7 +196,12 @@ export const DataGrid = function DataGrid<TData>({
                           };
 
                       const headerContent = isPinned ? (
-                        headerCell
+                        <Box
```

Comment:
@deniskaber This is the most controversial part of the PR. Basically we need this wrapper to add right-side border to the header cell of the row_id row